pkg/tools/sandbox/utils.go

```diff
@@ -60,29 +65,52 @@ func NewEndpoint(ctx context.Context, nse *registry.NetworkServiceEndpoint, gene
 		}
 	}
 	serve(ctx, u, ep.Register)
+
 	if nse.Url == "" {
 		nse.Url = u.String()
 	}
 	if nse.ExpirationTime == nil {
-		deadline := time.Now().Add(time.Hour)
-		expirationTime, err := ptypes.TimestampProto(deadline)
-		if err != nil {
-			return nil, err
-		}
-		nse.ExpirationTime = expirationTime
+		nse.ExpirationTime = timestamppb.New(time.Now().Add(time.Hour))
 	}
-	if _, err := mgr.NetworkServiceEndpointRegistryServer().Register(ctx, nse); err != nil {
+
+	var reg *registry.NetworkServiceEndpoint
+	if reg, err = mgr.NetworkServiceEndpointRegistryServer().Register(ctx, nse); err != nil {
 		return nil, err
 	}
+
+	nse.Name = reg.Name
+	nse.ExpirationTime = reg.ExpirationTime
+
 	for _, service := range nse.NetworkServiceNames {
-		if _, err := mgr.NetworkServiceRegistryServer().Register(ctx, &registry.NetworkService{Name: service, Payload: "IP"}); err != nil {
+		if _, err := mgr.NetworkServiceRegistryServer().Register(ctx, &registry.NetworkService{
+			Name:    service,
+			Payload: payload.IP,
+		}); err != nil {
 			return nil, err
 		}
 	}
+
 	logger.Log(ctx).Infof("Started listen endpoint %v on %v.", nse.Name, u.String())
+
 	return &EndpointEntry{Endpoint: ep, URL: u}, nil
 }
 
+// UnregisterEndpoint unregisters endpoint from NSMgr
+func UnregisterEndpoint(ctx context.Context, nse *registry.NetworkServiceEndpoint, mgr nsmgr.Nsmgr) (err error) {
```

Comment:
As I can see this func is using by one test. Can we make this incline?

Reply:
Yep, inlined.